extensions/default/src/DicomJSONDataSource/index.js

```diff
@@ -258,19 +262,19 @@ function createDicomJSONApi(dicomJsonConfig) {
 
       const { StudyInstanceUID, SeriesInstanceUID } = displaySet;
       const study = findStudies('StudyInstanceUID', StudyInstanceUID)[0];
-      const series = study.series.find(s => s.SeriesInstanceUID === SeriesInstanceUID);
-      let instances = displaySet.images;
-      if (series.instances.length > displaySet.images.length) {
-        instances = series.instances;
-      }
+      const series = study.series.find(s => s.SeriesInstanceUID === SeriesInstanceUID) || [];
 
-      instances.forEach(instance => {
+      displaySet.images.forEach(instance => {
         const NumberOfFrames = instance.NumberOfFrames;
-
         if (NumberOfFrames > 1) {
           for (let i = 0; i < NumberOfFrames; i++) {
+            /** In case there are multiple sop class we filter sop instances */
+            const images = series.instances
+              .filter(i => i && i.metadata && i.metadata.SOPInstanceUID === instance.SOPInstanceUID)
```

Comment:
Please rename the filter variable i to not override the loop variable i.
Also, 
.filter(testInstance => testInstance?.metadata?.SOPInstanceUID === instance.SOPInstanceUID
is easier to read than the && version.

Reply:
Updated!